### PR TITLE
[mentions] highlighted mentions in input

### PR DESCRIPTION
**Pull request by rasom**
needed for #11179

status: ready

### CODE REVIEW

**Comment by rasom** [comment]
@errorists can you take a look please and tell what's not working as expected here

**Comment by errorists** [comment]
@rasom other than this dude having a ridiculous username, I find nothing wrong on the screenshot

**Comment by rasom** [comment]
hey @churik @qoqobolo, can you please take a look at this one? It is needed for adding nicknames to suggestions. It is fine to be tested a bit before it gets reviewed because I need some feedback on ui/ux anyway.

Possible problems:
1. Different text insertions (inside already highlighted mentions, before the first char, at the end, etc)
2. Slow responsiveness on typing/removing 
3. Something else you find

**Comment by qoqobolo** [comment]
@rasom at the moment see this (probably) issue.

#### Issue 1: On iOS device can insert emoji into already highlighted mention and it is still highlighted.

Screenshot 2020-09-17 at 17 10 50

**Comment by churik** [comment]
@rasom 
### ISSUE 2: deleting @ at the beginning of mention erase all text in the input field
BAACAgQAAxkBAAErZUtfY35w84L_2NZsC4a4tbVvhfH4bQACNgkAAo42IFMH5I_cesKTLRsE_1600356023 49

**Comment by rasom** [comment]
@qoqobolo issue 1 looks like RN bug... I'll try to fix it but probably it would be just easier to ignore it for now

**Comment by rasom** [comment]
@churik @qoqobolo 
issue 2 is fixed 
issue 1 won't be fixed, that's RN bug

**Comment by churik** [comment]
@rasom noticed 2 issues:
#### ISSUE 3: all text before mention resolves as 0
#### ISSUE 4: Java exception at attempt to select mention while reproducing issue 3

Both issues are on this gif
BAACAgQAAxkBAAEra-VfZHiFG2AFBwNEy6SoCnRjw9zPugAChQgAAmoWIVPQhBnMqEXaPhsE_1600420194 39

**Comment by rasom** [comment]
@churik fixed issue 3, can't reproduce 4

**Review by hesterbruikman — APPROVED** [review]
Looks good @rasom!

**Comment by churik** [comment]
Tested on Android10 and IOS13:
- ENS names + contacts + chat attendees are suggested and highlighted in input
- moving caret back and forward the input with mentions
- mention of myself
- 10+mentions in input field
- mentions and different languages
- messages in replies, messages as comment for image

File: src/status_im/subs.cljs [hunk 1]
```diff
@@ -904,6 +904,13 @@
  (fn [[chat-id cursor]]
    (get cursor chat-id)))
 
+(re-frame/reg-sub
+ :chat/input-with-mentions
```
Comment (flexsurfer) [inline src/status_im/subs.cljs:908]: what the difference between :chat/input-with-mentions and :chat**S**/input-with-mentions?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/subs.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/subs.cljs:908]: `chats/input-with-mentions` is just a db map of `{chat-id data}` format, when `chat/input-with-mentions` is a chat specific `data`. I can rename it if that's confusing, just let me know what would look better here

**Pull request merged**